Shopify/cli
# Added external extension type to send new

### WHY are these changes introduced?

Fixes https://github.com/Shopify/shopify-cli-planning/issues/300

### WHAT is this pull request doing?
- Displays new internal extension type names everywhere they are shown in the cli
- Supports  new internal extension type names inside the `.toml` files and the old ones too
- Scaffold command type flags supports now new  new internal extension type names
- Dev command uses new extension binary attribute `external_type` so new names is displayed in the dev-console #https://github.com/Shopify/shopify-cli-extensions/pull/369
- Scaffold command uses new extension binary attribute `external_type` so new names is used in the `.toml` files #https://github.com/Shopify/shopify-cli-extensions/pull/369

### How to test your changes?
- Run scaffold command using the `-t` with a new UI extension type name. The `.toml` file generated should have the new name
- Run dev command. Open the browser and new names should be displayed in the type column. Preview the extension and in the browser inspecting the network response that obtains the extension information, the old extension type name should be returned
